modules/commons/src/main/frontend/src/components/FormattedText.jsx

```diff
@@ -0,0 +1,55 @@
+//
+//  Licensed to the Apache Software Foundation (ASF) under one
+//  or more contributor license agreements.  See the NOTICE file
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Typography, withStyles, makeStyles } from "@material-ui/core";
+import MDEditor from '@uiw/react-md-editor';
+
+const useStyles = makeStyles(theme => ({
+  markdown: {
+      fontSize: "inherit",
+  }
+}));
+
+let FormattedText = (props) => {
+  let { children, variant, display, color } = props;
+  const classes = useStyles();
+
+  return (
+    <Typography variant={variant} display={display} component="div" color={color} >
+      <MDEditor.Markdown className={classes.markdown} source={children} />
+    </Typography>
```

Comment:
Here's a suggestion to fix the "discarded styling" bug from the previous comment. `typographyProps` will include `classes` and `className` set when instantiating the `FormattedText` component. This also simplifies the implementation by sending all props that we might set on `FormattedText` directly to `Typography`. The use of the  `markdown` className is restricted only to the internal `MDEditor` component.

```suggestion
  let { children, ...typographyProps } = props;
  const mdClasses = useStyles();

  return (
    <Typography component="div" {...typographyProps} >
      <MDEditor.Markdown classes={mdClasses} className={mdClasses.markdown} source={children} />
    </Typography>
```